File: 1_images/scripts/my_subscriber.py
#!/usr/bin/env python
import rospy
from sensor_msgs.msg import Image
import cv2
from cv_bridge import CvBridge
import numpy as np
import traceback
import tf

#Declare your global variables
global img, is_img, display

img=[]
is_img = False

display = True

def callback_image(msg):
	global img, is_img

	try:
		#Transform the new message to an OpenCV matrix
		bridge_rgb=CvBridge()
		img = bridge_rgb.imgmsg_to_cv2(msg,msg.encoding).copy()
		is_img = True
	except:
		rospy.logerr(traceback.format_exc())

def main():
	#Start your ROS node
	rospy.loginfo('Starting image_listener application in python...')
	rospy.init_node('image_listener', anonymous=True)

	#Tell ROS what to do when a new message (Image) arrives!
	#ROS will execute the commands in the callback functions
	rospy.Subscriber("camera/image", Image , callback_image)

	global img, is_img, display

	loop=rospy.Rate(30)
	while not rospy.is_shutdown():
		if is_img:
			#Show your images
			if(display):
				cv2.imshow("view", img)
				cv2.waitKey(1)

			loop.sleep()

if __name__=='__main__':
	try:
		main()
		cv2.destroyAllWindows()
	except rospy.ROSInterruptException:
		pass

File: 1_images/src/my_subscriber.cpp
#include <ros/ros.h>
#include <image_transport/image_transport.h>
#include <cv_bridge/cv_bridge.h>

#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/highgui/highgui.hpp>

#include <string>

bool display = true;

void callback_image(const sensor_msgs::ImageConstPtr& msg)
{
	try
	{
		//Transform the new message to an OpenCV matrix
		cv::Mat img;
		img = cv_bridge::toCvShare(msg, "bgr8")->image.clone();

		//Show your images
		if (display)
		{
			cv::imshow("view", img);
			cv::waitKey(30);
		}
	}
	catch (cv_bridge::Exception& e)
	{
		//In case of error, report it
		ROS_ERROR("Could not convert from '%s' to 'bgr8'.", msg->encoding.c_str());
	}

}

int main(int argc, char **argv)
{
	//Start your ROS node
	ROS_INFO("Starting image_listener application in c++...");
	ros::init(argc, argv, "image_listener");

	//Create a handler for your ROS element
  	ros::NodeHandle nh;
	image_transport::ImageTransport it(nh);
	//Tell ROS what to do when a new message (Image) arrives!
	//ROS will execute the commands in the callback function
	image_transport::Subscriber sub = it.subscribe("camera/image", 1, callback_image);

	//Start an infinite loop in ROS
	ros::spin();

	//Destroy any opened OpenCV window before finishing 
	cv::destroyWindow("view");
}
